src/components/measureGroups/SupplementalElements.tsx

```diff
@@ -1,7 +1,294 @@
+    measure?.createdBy,
+    measure?.acls,
+    measure?.measureMetaData?.draft
+  );
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      supplementalDataElements: [],
+      descriptions: [],
+    },
+    onSubmit: (values) => {
+      submitForm(values);
+    },
+  });
+  const { resetForm } = formik;
+
+  useEffect(() => {
+    if (measure?.cql) {
+      const definitions = new CqlAntlr(measure.cql).parse()
+        .expressionDefinitions;
+      const defines = definitions
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map((opt, i) => opt.name.replace(/"/g, ""));
+      setSupplementalDataElementList(defines);
+    } else if (measure?.cql !== undefined) {
+      handleToast(
+        "danger",
+        `Please complete the CQL Editor process before continuing`,
+        true
+      );
+    }
+  }, [measure]);
+
+  const submitForm = (values) => {
+    const submitMeasure = {
+      ...measure,
+    };
+
+    //to-do: update measure with new supplemental data elements
+    measureServiceApi
+      .updateMeasure(submitMeasure)
+      .then(() => {
+        handleToast(
+          "success",
+          `Supplement Data Element Information Saved Successfully`,
+          true
+        );
+        updateMeasure(submitMeasure);
+      })
+      .catch(() => {
+        const message = `Error updating measure "${measure.measureName}"`;
+        setErrorMessage(message);
+
+        handleToast(
+          "danger",
+          `Error updating measure "${measure.measureName}"`,
+          true
+        );
+      });
+  };
+
+  useEffect(() => {
+    const subscription = measureStore.subscribe(setMeasure);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  // Route handle store will give warning message
+  // if form is dirty and user ties to navigate across SPAs
+  const { updateRouteHandlerState } = routeHandlerStore;
+  const [discardDialogOpen, setDiscardDialogOpen] = useState(false);
+
+  useEffect(() => {
+    updateRouteHandlerState({
+      canTravel: !formik.dirty,
+      pendingRoute: "",
+    });
+  }, [formik.dirty]);
+
+  const handleDesciptionChange = (field, value, index) => {
+    selectedSupplementalDataList.filter((selected) => {
+      if (selected === field) {
+        formik.setFieldValue(field, value);
+      }
+    });
+  };
+
+  return (
+    <form
+      id="measure-details-form"
+      onSubmit={formik.handleSubmit}
+      data-testid="supplemental-data-form"
+    >
+      <div className="content">
+        <div className="subTitle">
+          <h2>Supplemental Data</h2>
+          <div className="required">
+            <Typography
+              style={{ fontSize: 14, fontWeight: 300, fontFamily: "Rubik" }}
+            >
+              <span style={asterisk}>*</span>
+              Indicates required field
+            </Typography>
+          </div>
+        </div>
+        <div>
+          {supplementalDataElementList && (
```

Comment:
I believe this will always be true, and so we may not need this check

Reply:
@RohitKandimalla User might not have selected any supplemental data, so this condition hides the description section.

Reply:
As per my understanding this supplementalDataElementList, is to store the definitions from CQL, and doesn't store any user selected list. Correct me if I am wrong